### DISCUSSION TITLE
Making sense of the debug trace output from mutool

### DISCUSSION THREAD

**Discussion by mani2106** [pymupdf/PyMuPDF#907] [Q&A] [answered]
I stumbled onto [this](https://github.com/pymupdf/PyMuPDF/issues/736#issuecomment-735276876) conversation, where you mention how to get the debug trace from mutool, I could not make sense for most of attributes in the xml like `trm`, `bidi` and `wmode` for the xml extracted with my file, Do you have any reference to look up?

the following is a sample
```xml
<fill_text colorspace="DeviceRGB" color="1 1 1" matrix="1 0 0 -1 0 0">
<span font="AAAAAA+Arial" wmode="0" bidi="0" trm="8.95 0 0 8.95">
<g unicode="U+003c" glyph="1" x="90.65" y="-26.95"/>
<g unicode="s" glyph="2" x="95.8768" y="-26.95"/>
<g unicode="c" glyph="3" x="100.3518" y="-26.95"/>
<g unicode="r" glyph="4" x="104.8268" y="-26.95"/>
<g unicode="i" glyph="5" x="107.807147" y="-26.95"/>
<g unicode="p" glyph="6" x="109.794048" y="-26.95"/>
```

**Comment by JorjMcKie** [comment, marked as answer]
* wmode="0" - writing mode, 0 = horizontal
* bidi="0" - bidirectional text writing level, 0 = left to right
* trm="8.95 0 0 8.95" - parameters `a` through `d` of the text matrix. `a` = `d` = 8.95 is the fontsize, `b`= `c` = 0 means no rotation in this case. The remaining matrix parameters, `e`, `f` are the positions `x`, `y` of each glyph.

**Reply by mani2106** [reply]
Thanks for the explanation, I have searched for some references, is this mentioned in the documentation of [mutool](https://www.mupdf.com/docs/manual-mutool-trace.html)? because I could not find any.

**Reply by JorjMcKie** [reply]
No - I know that from source code inspection.
I myself do not know every detail of possible trace output. Especially when it comes to complex situations like hidden text or optional content or other conditional displays.
